app/src/main/kotlin/at/bitfire/davdroid/sync/RealSyncAdapter.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.sync
+
+import android.accounts.Account
+import android.accounts.AccountManager
+import android.content.AbstractThreadedSyncAdapter
+import android.content.ContentProviderClient
+import android.content.ContentResolver
+import android.content.Context
+import android.content.SyncResult
+import android.os.Build
+import android.os.Bundle
+import android.os.IBinder
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import at.bitfire.davdroid.R
+import at.bitfire.davdroid.repository.DavCollectionRepository
+import at.bitfire.davdroid.repository.DavServiceRepository
+import at.bitfire.davdroid.resource.LocalAddressBook
+import at.bitfire.davdroid.settings.AccountSettings
+import at.bitfire.davdroid.sync.account.InvalidAccountException
+import at.bitfire.davdroid.sync.worker.BaseSyncWorker
+import at.bitfire.davdroid.sync.worker.SyncWorkerManager
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import java.util.logging.Level
+import java.util.logging.Logger
+import javax.inject.Inject
+
+/**
+ * Entry point for the Sync Adapter Framework.
+ *
+ * Handles incoming sync requests from the Sync Adapter Framework.
+ *
+ * Although we do not use the sync adapter for syncing anymore, we keep this sole
+ * adapter to provide exported services, which allow android system components and calendar,
+ * contacts or task apps to sync via DAVx5.
+ *
+ * All Sync Adapter Framework related interaction should happen inside [SyncFrameworkIntegration].
+ */
+class RealSyncAdapter @Inject constructor(
```

Comment:
Calling the fake sync adapter for "FakeSyncAdapter" in the tests makes sense, but why call the real one "RealSyncAdapter" ? It's quite clear that this is not a fake sync adapter if it's in the codebase (not in tests) or not?

Reply:
hmm, no. Not really. I guess we can leave it as is for now. Maybe we find another name at some point :)